resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -216,8 +216,70 @@ private[spark] class Client(
                 .endContainer()
               .endSpec()
             .done()
-          submitCompletedFuture.get(30, TimeUnit.SECONDS)
-        }
+          try {
+            submitCompletedFuture.get(30, TimeUnit.SECONDS)
```

Comment:
IIRC, in the spark standalone mode, as long as the master is in "ALIVE" state, the driver is queued -  translating to fire and forget. 

@foxish's concern is a valid one, in that scenario, the ideal thing is for the job/driver to be queued, though it might be gotten to at a later time. Perhaps, we can settle for an approach where the submission client polls for the driver state ?